pkg/controller/kappnav/kappnav_controller.go

```diff
@@ -458,13 +584,46 @@ func (r *ReconcileKappnav) Reconcile(request reconcile.Request) (reconcile.Resul
 		return nil
 	})
 	if err != nil {
-		reqLogger.Error(err, "Failed to reconcile the Controller Deployment")
+		if (logger.IsEnabled(kappnavutils.LogTypeError)) {
+			logger.Log(kappnavutils.CallerName(), kappnavutils.LogTypeError, fmt.Sprintf("Failed to reconcile the Controller Deployment" + otherLogData + ", Error: %s", err), logName)
+		}
 		return r.ManageError(err, kappnavv1.StatusConditionTypeReconciled, instance)
-	}
+	}	
 
 	// If an extension exists call its reconcile function, otherwise return success.
 	if extension != nil {
 		return extension.ReconcileAdditionalResources(request, &r.ReconcilerBase, instance)
 	}
 	return r.ManageSuccess(kappnavv1.StatusConditionTypeReconciled, instance)
+
+	
+}
+
+
+// get logging info from CR and set logger level
+func setLoggingLevel(loginfo string) {
+	if len(loginfo) == 0  || loginfo == "info" {
```

Comment:
if/else or switch statement (https://tour.golang.org/flowcontrol/9)?. not a big deal performance wise but all of those if statements will always be hit.